### PR TITLE
Remove reference to tests file in prod code

### PR DESCRIPTION
**Pull request by callegarimattia**
- Delete the xml after the loading is done
- Fixed coverage on tests folder
- Removed test file check on prod code

@krishnasism what do you think? Just an idea, but I do think we shouldn't have references to tests files in the prod code.
This may be a solution :)

### CODE REVIEW

**Comment by krishnasism** [comment]
Thanks for this - looks good!

**Comment by krishnasism** [comment]
Just fix the typo xlm to xml and merge this in

**Review by krishnasism — APPROVED** [review]

File: tests/test_functions.py [hunk 1]
```diff
@@ -19,6 +20,11 @@ def mock_hotpdf_bank_file_name():
     return "tests/resources/hotpdf_bank.pdf"
 
 
+def xlm_copy_file_name(xml_file_name: str):
```
Comment (callegarimattia) [inline tests/test_functions.py:23]:
```suggestion
def xml_copy_file_name(xml_file_name: str):
```

File: tests/test_functions.py [hunk 2]
```diff
@@ -98,24 +104,25 @@ def test_sparse_matrix_iterator():
...
 
 
-@patch("hotpdf.processor.generate_xml_file", return_value="tests/resources/xml/hotpdf_bank_dup_span.xml", autospec=True)
-def test_duplicate_spans_not_removed(_, mock_hotpdf_bank_file_name):
+def test_duplicate_spans_not_removed(mock_hotpdf_bank_file_name):
     hot_pdf_object = HotPdf()
     hot_pdf_object_with_dup_span = HotPdf()
-    hot_pdf_object_with_dup_span.load(mock_hotpdf_bank_file_name, drop_duplicate_spans=False)
-    hot_pdf_object.load(mock_hotpdf_bank_file_name)
+    with patch("hotpdf.processor.generate_xml_file", return_value=xlm_copy_file_name("tests/resources/xml/hotpdf_bank_dup_span.xml")):
```
Comment (callegarimattia) [inline tests/test_functions.py:110]:
```suggestion
    with patch("hotpdf.processor.generate_xml_file", return_value=xml_copy_file_name("tests/resources/xml/hotpdf_bank_dup_span.xml")):
```

Comment (callegarimattia) [inline tests/test_functions.py:110]:
```suggestion
    with patch("hotpdf.processor.generate_xml_file", return_value=xml_copy_file_name("tests/resources/xml/hotpdf_bank_dup_span.xml")):
```

File: tests/test_functions.py [hunk 3]
```diff
@@ -98,24 +104,25 @@ def test_sparse_matrix_iterator():
...
+        hot_pdf_object.load(mock_hotpdf_bank_file_name)
 
     assert len(hot_pdf_object.pages[0].span_map) < len(hot_pdf_object_with_dup_span.pages[0].span_map)
 
 
 def test_load_negative_coordinates(mock_hotpdf_bank_file_name):
     QUERY = "HOTPDF BANK"
-    with patch("hotpdf.processor.generate_xml_file", return_value="tests/resources/xml/hotpdf_bank_negative_coords.xml"):
+    with patch("hotpdf.processor.generate_xml_file", return_value=xlm_copy_file_name("tests/resources/xml/hotpdf_bank_negative_coords.xml")):
```
Comment (callegarimattia) [inline tests/test_functions.py:120]:
```suggestion
    with patch("hotpdf.processor.generate_xml_file", return_value=xml_copy_file_name("tests/resources/xml/hotpdf_bank_negative_coords.xml")):
```

File: tests/test_functions.py [hunk 4]
```diff
@@ -98,24 +104,25 @@ def test_sparse_matrix_iterator():
...
-@patch("hotpdf.processor.generate_xml_file", return_value="tests/resources/xml/hotpdf_bank_dup_span.xml", autospec=True)
-def test_duplicate_spans_not_removed(_, mock_hotpdf_bank_file_name):
+def test_duplicate_spans_not_removed(mock_hotpdf_bank_file_name):
     hot_pdf_object = HotPdf()
     hot_pdf_object_with_dup_span = HotPdf()
-    hot_pdf_object_with_dup_span.load(mock_hotpdf_bank_file_name, drop_duplicate_spans=False)
-    hot_pdf_object.load(mock_hotpdf_bank_file_name)
+    with patch("hotpdf.processor.generate_xml_file", return_value=xlm_copy_file_name("tests/resources/xml/hotpdf_bank_dup_span.xml")):
+        hot_pdf_object_with_dup_span.load(mock_hotpdf_bank_file_name, drop_duplicate_spans=False)
+    with patch("hotpdf.processor.generate_xml_file", return_value=xlm_copy_file_name("tests/resources/xml/hotpdf_bank_dup_span.xml")):
```
Comment (callegarimattia) [inline tests/test_functions.py:112]:
```suggestion
    with patch("hotpdf.processor.generate_xml_file", return_value=xml_copy_file_name("tests/resources/xml/hotpdf_bank_dup_span.xml")):
```

File: tests/test_functions.py [hunk 5]
```diff
@@ -98,24 +104,25 @@ def test_sparse_matrix_iterator():
...
 def test_load_negative_coordinates(mock_hotpdf_bank_file_name):
     QUERY = "HOTPDF BANK"
-    with patch("hotpdf.processor.generate_xml_file", return_value="tests/resources/xml/hotpdf_bank_negative_coords.xml"):
+    with patch("hotpdf.processor.generate_xml_file", return_value=xlm_copy_file_name("tests/resources/xml/hotpdf_bank_negative_coords.xml")):
         hot_pdf_object = HotPdf()
         hot_pdf_object.load(mock_hotpdf_bank_file_name)
         assert not hot_pdf_object.find_text(QUERY)[0], "Expected string to be empty"
     # For sanity: The following file is same as above, except the coords are normal
-    with patch("hotpdf.processor.generate_xml_file", return_value="tests/resources/xml/hotpdf_bank_normal_coords.xml"):
+    with patch("hotpdf.processor.generate_xml_file", return_value=xlm_copy_file_name("tests/resources/xml/hotpdf_bank_normal_coords.xml")):
```
Comment (callegarimattia) [inline tests/test_functions.py:125]:
```suggestion
    with patch("hotpdf.processor.generate_xml_file", return_value=xml_copy_file_name("tests/resources/xml/hotpdf_bank_normal_coords.xml")):
```

**Review by callegarimattia — COMMENTED** [review]
Thanks for spotting the typo :)

**Pull request merged**